js/src/setup-mc/setup-stepper/setup-free-listings/shipping-rate/shipping-rate-setup/useShippingRatesWithSavedSuggestions.test.js

```diff
@@ -0,0 +1,148 @@
+
+const shippingRatesData = [
+	{
+		country: 'Malaysia',
+		country_code: 'MY',
+		currency: 'USD',
+		rate: '20',
+	},
+];
+
+const shippingRatesSuggestionsData = [
+	{
+		country: 'Malaysia',
+		country_code: 'MY',
+		currency: 'USD',
+		rate: 20,
+	},
+];
+
+describe( 'useShippingRatesWithSavedSuggestions', () => {
+	it( 'should save suggestions as shipping rates when initial shipping rates is empty', async () => {
+		useShippingRates
+			.mockReturnValueOnce( {
+				hasFinishedResolution: false,
+				data: undefined,
+			} )
+			.mockReturnValue( {
+				hasFinishedResolution: true,
+				data: [],
+			} );
+		useShippingRatesSuggestions
+			.mockReturnValueOnce( {
+				loading: true,
+				data: undefined,
+			} )
+			.mockReturnValueOnce( {
+				loading: true,
+				data: undefined,
+			} )
+			.mockReturnValue( {
+				loading: false,
+				data: shippingRatesSuggestionsData,
+			} );
+		const mockSaveSuggestions = jest.fn();
+		useSaveSuggestions.mockReturnValue( mockSaveSuggestions );
+
+		const { result, rerender, waitForNextUpdate } = renderHook( () =>
+			useShippingRatesWithSavedSuggestions()
+		);
+
+		/**
+		 * Shipping rates and suggestions are loading.
+		 */
+		expect( result.current.loading ).toBe( true );
+		expect( result.current.data ).toBe( undefined );
+		expect( mockSaveSuggestions ).toHaveBeenCalledTimes( 0 );
+
+		/**
+		 * Shipping rates are loaded; suggestions are loading.
+		 */
+		rerender();
+		expect( result.current.loading ).toBe( true );
+		expect( result.current.data ).toStrictEqual( [] );
+		expect( mockSaveSuggestions ).toHaveBeenCalledTimes( 0 );
+
+		/**
+		 * Shipping rates and suggestions are loaded,
+		 * and saveSuggestions is called.
+		 */
+		rerender();
+		await waitForNextUpdate();
+		expect( result.current.loading ).toBe( false );
+		expect( result.current.data ).toStrictEqual( [] );
+		expect( mockSaveSuggestions ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	it( 'should not save suggestions as shipping rates when there is an initial shipping rates', async () => {
+		useShippingRates
+			.mockReturnValueOnce( {
+				hasFinishedResolution: false,
+				data: undefined,
+			} )
+			.mockReturnValue( {
+				hasFinishedResolution: true,
+				data: shippingRatesData,
+			} );
+		useShippingRatesSuggestions
+			.mockReturnValueOnce( {
+				loading: true,
+				data: undefined,
+			} )
+			.mockReturnValueOnce( {
+				loading: true,
+				data: undefined,
+			} )
+			.mockReturnValue( {
+				loading: false,
+				data: shippingRatesSuggestionsData,
+			} );
+		const mockSaveSuggestions = jest.fn();
+		useSaveSuggestions.mockReturnValue( mockSaveSuggestions );
+
+		const { result, rerender } = renderHook( () =>
+			useShippingRatesWithSavedSuggestions()
+		);
+
+		/**
+		 * Shipping rates and suggestions are loading.
+		 */
+		expect( result.current.loading ).toBe( true );
+		expect( result.current.data ).toBe( undefined );
+		expect( mockSaveSuggestions ).toHaveBeenCalledTimes( 0 );
+
+		// rerender();
```

Comment:
💅  I guess this is not necessary here

Reply:
Fixed in [`8b95fc9` (#1161)](https://github.com/woocommerce/google-listings-and-ads/pull/1161/commits/8b95fc9209c6d4bac51a73a6f85a3c54fa400ca0).